packages/amplify_core/test/state_machine/dependency_manager_test.dart

```diff
@@ -67,40 +102,125 @@ void main() {
         () => dependencyManager.getOrCreate<MyDependency>(),
         throwsStateError,
       );
+      expect(
+        () => dependencyManager.expect<MyDependency>(),
+        throwsStateError,
+      );
+      expect(
+        () => scopedDependencyManager.get<MyDependency>(),
+        returnsNormally,
+      );
+      expect(
+        () => scopedDependencyManager.getOrCreate<MyDependency>(),
+        throwsStateError,
+      );
+      expect(
+        () => scopedDependencyManager.expect<MyDependency>(),
+        throwsStateError,
+      );
     });
 
     test('injects dependency manager', () {
-      const token = Token<NeedsDependencyManager>([
-        Token<DependencyManager>(),
-      ]);
       dependencyManager.addBuilder<NeedsDependencyManager>(
         NeedsDependencyManager.new,
-        token,
       );
 
       expect(
-        () => dependencyManager.getOrCreate<NeedsDependencyManager>(token),
+        () => dependencyManager.getOrCreate<NeedsDependencyManager>(),
+        returnsNormally,
+      );
+      expect(
+        () => scopedDependencyManager.getOrCreate<NeedsDependencyManager>(),
         returnsNormally,
       );
     });
 
     test('injects dependency manager and dispatcher', () {
-      const token = Token<NeedsDependencyManagerAndDispatcher>([
-        Token<DependencyManager>(),
-        Token<Dispatcher>(),
-      ]);
-      dependencyManager
-        ..addBuilder<Dispatcher>(MyDispatcher.new)
-        ..addBuilder<NeedsDependencyManagerAndDispatcher>(
-          NeedsDependencyManagerAndDispatcher.new,
-          token,
-        );
+      dependencyManager.addBuilder<Dispatcher>((_) => MyDispatcher());
+      dependencyManager.addBuilder<NeedsDependencyManagerAndDispatcher>(
+        (deps) => NeedsDependencyManagerAndDispatcher(
+          deps,
+          deps.getOrCreate(),
+        ),
+      );
 
       expect(
         () => dependencyManager
-            .getOrCreate<NeedsDependencyManagerAndDispatcher>(token),
+            .getOrCreate<NeedsDependencyManagerAndDispatcher>(),
+        returnsNormally,
+      );
+      expect(
+        () => scopedDependencyManager
+            .getOrCreate<NeedsDependencyManagerAndDispatcher>(),
+        returnsNormally,
+      );
+    });
+
+    test('scoping does not affect parent', () {
+      scopedDependencyManager.addBuilder<NeedsDependencyManager>(
+        NeedsDependencyManager.new,
+      );
+
+      expect(
+        () => scopedDependencyManager.getOrCreate<NeedsDependencyManager>(),
         returnsNormally,
       );
+      expect(
+        () => dependencyManager.getOrCreate<NeedsDependencyManager>(),
+        throwsStateError,
+      );
+      expect(dependencyManager.get<DependencyManager>(), dependencyManager);
+      expect(
+        scopedDependencyManager.get<DependencyManager>(),
+        scopedDependencyManager,
+      );
+    });
+
+    test('create closes previous instances', () {
```

Comment:
Not sure the test description fits the test. Should this test call `scopedDependencyManager.create` a second time and then assert that `scopedDep` is closed?

Reply:
Good catch - not sure what I was thinking. Will update to match name!